torch/testing/_internal/common_modules.py

```diff
@@ -247,6 +247,23 @@ def no_batch_dim_reference_criterion_fn(m, *args, **kwargs):
     return output
 
 
+def no_batch_dim_reference_mha(m, p, *args, **kwargs):
```

Comment:
I wonder if there's some way to make this more generic? We'll need similar logic involving `batch_first` for the various RNNs.

Going down this route, I think we could make this fully generic if we had some reference function that was configurable in these ways:
* Accepts which inputs to squeeze / unsqueeze (both args and kwargs)
* Accepts which outputs to squeeze / unsqueeze
* Handles `batch_first` (for sequence-based modules e.g. MHA / RNNs)
* Handles `reduction` (for criterion modules)

wdyt about this?

Reply:
I think MHA has a very unique structure
Eg. output having batch-dim at different position.
```
(output[0].squeeze(dim), output[1].squeeze(0))
```

I think instead of trying to write a complicated generic function we might keep it seperate for now. Going forward with RNNs and other Transformer related layer we can introduce the generic version.

Reply:
> I think MHA has a very unique structure
Eg. output having batch-dim at different position.

This is true and it does make it tricky. I'm good with leaving it in place for now, but I hope there will be a consolidation opportunity for the various RNN types.